src/lib/Imp.hs

```diff
@@ -466,10 +473,22 @@ toImpOp maybeDest op = case op of
       SumCon    _ tag payload -> Variant labs "c" tag payload
       SumAsProd _ tag payload -> Con $ SumAsProd resultTy tag payload
       _ -> error $ "Not a sum type: " ++ pprint (Con c)
+  -- Listing branches that should be dead helps GHC cut down on code size.
```

Comment:
Well otherwise it will specialize the `_ -> ...` case to those particular constructors, since it has no way of knowing statically that they won't appear here. So we can help it like that!